tests/virt_autotest/virt_utils.pm

```diff
@@ -103,10 +108,39 @@ sub repl_guest_autoyast_addon_with_daily_build_module {
     repl_addon_with_daily_build_module_in_files("$file_list");
 }
 
+#many virtualization testsuites contain settings composed by %DISTRI%s-%VERSION%
+#that need special handling for sp0/fcs, when integer products are being tested, eg sle15
+sub handle_sp_in_settings {
+    my ($var_name, $value_for_sp) = @_;
+
+    die "We only support sles product currently!" unless (get_required_var('DISTRI') eq 'sle');
+
+    #We need small case variable value
+    my $var_value = lc(get_required_var("$var_name"));
+    #Add $value_for_sp after release for products like sle15, sle16
+    if ($var_value !~ /sp/ && "$value_for_sp") {
+        $var_value =~ s/(sles-\d+)/$1-$value_for_sp/;
+    }
+    set_var("$var_name", "$var_value");
+    bmwqemu::save_vars();
```

Comment:
I actually give the reason in the background :)

Reason is we call test scripts from openqa, and the parameters for calling those scripts are partially set as settings for testsuite. For example for guest installation, we need GUEST setting to refer to which guest to test, and the underlying script needs format like sles-12-sp4 or sles-15-fcs, however to be general, the openqa testsuite setting is %DISTRI%s-%VERSION%, of course it can not satisfy all situations. So we need special handling for such parameters.